avm/res/operational-insights/workspace/main.bicep

```diff
@@ -0,0 +1,413 @@
+    functionParameters: contains(savedSearch, 'functionParameters') ? savedSearch.functionParameters : ''
+    version: contains(savedSearch, 'version') ? savedSearch.version : 2
+  }
+  dependsOn: [
+    logAnalyticsWorkspace_linkedStorageAccounts
+  ]
+}]
+
+module logAnalyticsWorkspace_dataExports 'data-export/main.bicep' = [for (dataExport, index) in dataExports: {
+  name: '${uniqueString(deployment().name, location)}-LAW-DataExport-${index}'
+  params: {
+    workspaceName: logAnalyticsWorkspace.name
+    name: dataExport.name
+    destination: contains(dataExport, 'destination') ? dataExport.destination : {}
+    enable: contains(dataExport, 'enable') ? dataExport.enable : false
+    tableNames: contains(dataExport, 'tableNames') ? dataExport.tableNames : []
+  }
+}]
+
+module logAnalyticsWorkspace_dataSources 'data-source/main.bicep' = [for (dataSource, index) in dataSources: {
+  name: '${uniqueString(deployment().name, location)}-LAW-DataSource-${index}'
+  params: {
+    logAnalyticsWorkspaceName: logAnalyticsWorkspace.name
+    name: dataSource.name
+    kind: dataSource.kind
+    linkedResourceId: contains(dataSource, 'linkedResourceId') ? dataSource.linkedResourceId : ''
+    eventLogName: contains(dataSource, 'eventLogName') ? dataSource.eventLogName : ''
+    eventTypes: contains(dataSource, 'eventTypes') ? dataSource.eventTypes : []
+    objectName: contains(dataSource, 'objectName') ? dataSource.objectName : ''
+    instanceName: contains(dataSource, 'instanceName') ? dataSource.instanceName : ''
+    intervalSeconds: contains(dataSource, 'intervalSeconds') ? dataSource.intervalSeconds : 60
+    counterName: contains(dataSource, 'counterName') ? dataSource.counterName : ''
+    state: contains(dataSource, 'state') ? dataSource.state : ''
+    syslogName: contains(dataSource, 'syslogName') ? dataSource.syslogName : ''
+    syslogSeverities: contains(dataSource, 'syslogSeverities') ? dataSource.syslogSeverities : []
+    performanceCounters: contains(dataSource, 'performanceCounters') ? dataSource.performanceCounters : []
+  }
+}]
+
+module logAnalyticsWorkspace_tables 'table/main.bicep' = [for (table, index) in tables: {
+  name: '${uniqueString(deployment().name, location)}-LAW-Table-${index}'
+  params: {
+    workspaceName: logAnalyticsWorkspace.name
+    name: table.name
+    plan: contains(table, 'plan') ? table.plan : 'Analytics'
+    schema: contains(table, 'schema') ? table.schema : {}
+    retentionInDays: contains(table, 'retentionInDays') ? table.retentionInDays : -1
+    totalRetentionInDays: contains(table, 'totalRetentionInDays') ? table.totalRetentionInDays : -1
+    restoredLogs: contains(table, 'restoredLogs') ? table.restoredLogs : {}
+    searchResults: contains(table, 'searchResults') ? table.searchResults : {}
+  }
+}]
+
+module logAnalyticsWorkspace_solutions '../../operations-management/solution/main.bicep' = [for (gallerySolution, index) in gallerySolutions: if (!empty(gallerySolutions)) {
```

Comment:
Should reference the published module and requires a regeneration of the files

Reply:
Done